overlay.go

```diff
@@ -70,10 +70,16 @@ import (
 // data and can thus be safely manipulated by the caller.
 func ExtractBaseAndOverlayParts(bd *BundleData) (base, overlay *BundleData, err error) {
 	base = cloneBundleData(bd)
-	_ = visitField(&visitorContext{structVisitor: clearOverlayFields}, base)
+	_ = visitField(&visitorContext{
+		structVisitor:          clearOverlayFields,
+		dropNonRequiredMapKeys: false,
```

Comment:
By virtue of copy-paste I had initially set this to true which is incorrect. It can actually clear the contents of maps such as `options` so it is only required for the overlay bit.

I have now fixed this and added an option block to the relevant test to verify that it works as expected for the base bundle.